File: moudle-calendar-schedule/src/main/java/com/marshal/moudle/calendar/schedule/SizeUtils.java
package com.marshal.moudle.calendar.schedule;
import android.content.Context;
import android.util.DisplayMetrics;
import android.view.WindowManager;


public class SizeUtils {

    /**
     * 获取高度
     * @param context
     * @return
     */
    public static int  getScreenHeight(Context context) {
        WindowManager windowManager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
        DisplayMetrics displayMetrics = new DisplayMetrics();
        windowManager.getDefaultDisplay().getMetrics(displayMetrics);
        return displayMetrics.heightPixels;
    }

    public static int getDisplayMetrics(Context context){
        DisplayMetrics displayMetrics = new DisplayMetrics();
        WindowManager windowManager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
        windowManager.getDefaultDisplay().getMetrics(displayMetrics);
        float density = displayMetrics.density;
        // 将密度转换为 dpi (每英寸点数)
        return (int) (density * 160);
    }

    public static int dip2px(Context context, float dpValue) {
        float scale = context.getResources().getDisplayMetrics().density;
        return ((int) (dpValue * scale + 0.5f));
    }

    public static int px2dip(Context context, float pxValue) {
        float scale = context.getResources().getDisplayMetrics().density;
        return ((int) (pxValue / scale + 0.5f));
    }


}

File: base-common/src/main/java/com/marshal/base_common/utils/SizeUtils.kt
import android.content.Context
import android.util.DisplayMetrics
import android.view.WindowManager


object SizeUtils {

    /**
     * 根据手机的分辨率从 dp 的单位 转成为 px(像素)
     */
    fun dip2px(context: Context?, dpValue: Float): Int {
        val scale: Float = context?.resources?.displayMetrics?.density ?: 0f
        return ((dpValue * scale + 0.5f).toInt())
    }


    /**
     * 根据手机的分辨率从 px(像素) 的单位 转成为 dp
     */
    fun px2dip(context: Context?, pxValue: Float): Int {
        val scale: Float = context?.resources?.displayMetrics?.density ?: 0f
        return ((pxValue / scale + 0.5f).toInt())
    }

    fun getScreenHeight(context: Context?): Int {
        val displayMetrics = DisplayMetrics()
        (context?.getSystemService(Context.WINDOW_SERVICE) as WindowManager).defaultDisplay?.getMetrics(
            displayMetrics
        )
        return displayMetrics.heightPixels
    }


}